src/main/java/org/intellij/sdk/codesync/clients/CodeSyncClient.java

```diff
@@ -105,6 +105,7 @@ public Integer uploadFile(String accessToken, ConfigRepo configRepo, DiffFile di
 
         payload.put("repo_id", configRepo.id);
         payload.put("branch", diffFile.branch);
+        payload.put("commit_hash", diffFile.commitHash);
```

Comment:
Interesting, I don't think server-endpoint to upload file accepts this field since we currently have `commit_hash` only in `file_history` table. But I was thinking it should also go in `Files` table so that we know New Files related to given commit hash.

I was also thinking of having a separate table as 
```
id. file_id, commit_hash
```
There will be data redundancy since we can get this info from `file_history` table but some of quries will be faster since we won't be looking in file_history table (if possible)
This table will be populated from following endpoints
```
- POST /v1/init
- POST /v1/files
- WEBSOCKET /v2/websocket
```
Let me know your thoughts

Reply:
I agree that a bit of redundancy is okay to avoid querying the large table as `file_history` 👍🏼